Assets/MRTK/Services/InputSystem/MixedRealityInputModule.cs

```diff
@@ -305,8 +305,28 @@ protected PointerEventData.FramePressState StateForPointer(PointerData pointerDa
         void IMixedRealityPointerHandler.OnPointerUp(MixedRealityPointerEventData eventData)
         {
             int pointerId = (int)eventData.Pointer.PointerId;
-            Debug.Assert(pointerDataToUpdate.ContainsKey(pointerId));
-            pointerDataToUpdate[pointerId].nextPressState = PointerEventData.FramePressState.Released;
+
+            // OnPointerUp can be raised during an OnSourceLost. If the pointer has already been removed
+            // from the pointerDataToUpdate Dictionary and added to the pointerDataToRemove list, then
+            // that pointer's state update can be ignored.
+            Debug.Assert(pointerDataToUpdate.ContainsKey(pointerId) || IsPointerIdInRemovedList());
+            if (pointerDataToUpdate.TryGetValue(pointerId, out PointerData pointerData))
+            {
+                pointerData.nextPressState = PointerEventData.FramePressState.Released;
+            }
+
+            bool IsPointerIdInRemovedList()
```

Comment:
Nit: I don't think that we currently use a lot of local functions in this codebase - I think that I'd like to have this be a 'regular' private (i.e. non local) for now so that we can maybe have a discussion first on what we think about this).

Reply:
I just updated this in the fork, gonna push this in.